=====
Gingr
=====

.. image:: gingr/logo_med.png

Gingr is an interactive tool for exploring large-scale phylogenies in tandem with their corresponding multi-alignments. Gingr can display informative overviews for hundreds or thousands of genomes, while allowing researchers to move quickly to more detailed views of specific subclades and genomic regions, even down to the nucleotide level of their multi-alignments. Additionally, its dynamic display of variants allows interactive selection of various filters, such as indels, poorly aligned regions and suspected sites of recombination. Gingr works chiefly in tandem with Parsnp, an efficient tool for core-genome multi-alignment and phylogenetic reconstruction. It is also applicable, however, to other analytical tools, accepting standard file formats such as multi-Fasta, XMFA, Newick and VCF.  

.. image:: gingr/screen.png
   :width: 462
   :height: 290

**Download (v1.2)**

 * `gingr-OSX64-v1.2.zip <https://github.com/marbl/gingr/releases/download/v1.0.1/gingr-OSX64-v1.2.zip/>`_
 * `gingr-Linux64-v1.2.tar.gz <https://github.com/marbl/harvest/releases/download/v1.2/harvest-Linux64-v1.2.tar.gz/>`_

**Documentation**

.. toctree::
   :maxdepth: 2

   gingr/requirements
   gingr/tutorial
   gingr/types

**Resources**

`All releases`_ ~ `Source code`_ ~ `Report an issue`_

.. _All releases: https://github.com/marbl/gingr/releases
.. _Source code: https://github.com/marbl/gingr
.. _Report an issue: https://github.com/marbl/gingr/issues

